HoneyDo.Web/ClientApp/src/components/Tasks/Form.tsx

```diff
@@ -34,7 +53,8 @@ class TaskForm extends Component<TaskFormProps, TaskFormState> {
   constructor(props: TaskFormProps) {
     super(props);
     const name = (props.task && props.task.name) || "";
-    this.state = { name, hasError: false };
+    const dueDate = (props.task && props.task.dueDate) || undefined;
+    this.state = { name, hasError: false, dueDate };
```

Comment:
A cleaner way would be to do this FWIW:
```js
this.state = {
  name: props.name && props.task.name || "",
  dueDate: props.task && props.task.dueDate || null,
  hasError: false
}
```
or use proptypes:
```js
static defaultProps = {
  name: "",
  dueDate: null, // or don't specify it
  hasError: false
}
```
then you can do this:
```js
this.state = {
  name: props.name && props.task.name,
  dueDate: props.task && props.task.dueDate,
  hasError: false
}
```

Reply:
I don’t see that as cleaner, lol.